src/main/java/network/brightspots/rcv/ContestConfig.java

```diff
@@ -703,45 +701,52 @@ && getMaxRankingsAllowed() < MIN_MAX_RANKINGS_ALLOWED)) {
         MIN_MULTI_SEAT_BOTTOMS_UP_PERCENTAGE_THRESHOLD,
         MAX_MULTI_SEAT_BOTTOMS_UP_PERCENTAGE_THRESHOLD,
         false)) {
-      isValid = false;
+      validationErrors.add(
+          ValidationError.RULES_MULTI_SEAT_BOTTOMS_UP_PERCENTAGE_THRESHOLD_INVALID);
     }
 
     WinnerElectionMode winnerMode = getWinnerElectionMode();
     if (Utils.isInt(getNumberOfWinnersRaw())) {
       if (getNumberOfWinners() > 0) {
         if (isMultiSeatBottomsUpWithThresholdEnabled()) {
-          isValid = false;
+          validationErrors.add(
+              ValidationError.RULES_NUMBER_OF_WINNERS_DOES_INVALID_FOR_WINNER_ELECTION_MODE);
           Logger.severe(
               "numberOfWinners must be zero if winnerElectionMode is \"%s\"!", winnerMode);
         }
 
         if (getNumberOfWinners() > 1) {
           if (isContinueUntilTwoCandidatesRemainEnabled()) {
-            isValid = false;
+            validationErrors.add(
+                ValidationError.RULES_CONTINUE_UNTIL_TWO_CANDIDATES_REMAIN_TRUE_FOR_MULTI_SEAT);
             Logger.severe(
                 "continueUntilTwoCandidatesRemain can't be true in a multi-seat contest!");
           }
 
           if (isBatchEliminationEnabled()) {
-            isValid = false;
+            validationErrors.add(ValidationError.RULES_BATCH_ELIMINATION_TRUE_FOR_MULTI_SEAT);
             Logger.severe("batchElimination can't be true in a multi-seat contest!");
           }
         } else { // numberOfWinners == 1
           if (!isSingleWinnerEnabled()) {
-            isValid = false;
+            validationErrors.add(
+                ValidationError.RULES_WINNER_ELECTION_MODE_INVALID_FOR_SINGLE_SEAT);
             Logger.severe(
                 "winnerElectionMode can't be \"%s\" in a single-seat contest!", winnerMode);
           }
         }
       } else { // numberOfWinners == 0
         if (!isMultiSeatBottomsUpWithThresholdEnabled()) {
-          isValid = false;
+          validationErrors.add(
+              ValidationError
+                  .RULES_NUMBER_OF_WINNERS_WINNER_ELECTION_MODE_PERCENTAGE_THRESHOLD_DISAGREEMENT);
           Logger.severe(
               "If numberOfWinners is zero, winnerElectionMode must be \"%s\" and "
                   + "multiSeatBottomsUpPercentageThreshold must be specified!",
               WinnerElectionMode.MULTI_SEAT_BOTTOMS_UP_USING_PERCENTAGE_THRESHOLD);
         } else if (getMultiSeatBottomsUpPercentageThreshold() == null) {
```

Comment:
If numWinners is zero, two things must be true:
- `isMultiSeatBottomsUpWithThresholdEnabled()`
- `getMultiSeatBottomsUpPercentageThreshold !== null`

So I would just check each of these conditions independently here. It's a little confusing because the chain of requirements is:
zero winners --> multi-seat bottoms-up --> non-null threshold

But I would phrase the first check as "zero winners --> must use multi-seat bottoms-up" and the second check as "zero winners --> must set threshold."

This is a very minor thing, though, so do what you think is best.

Reply:
That makes sense, thanks! Changed it.